core/container.go

```diff
@@ -927,20 +927,37 @@ func (s *containerSchema) withoutVariable(ctx *router.Context, parent *Container
 	})
 }
 
-func (s *containerSchema) variables(ctx *router.Context, parent *Container, args any) ([]string, error) {
+type EnvVariable struct {
+	Name  string
+	Value string
```

Comment:
shot in the dark: try adding `json` tags to these to serialize them to `name` and `value`. I think (could be wrong) without that this will get serialized to "Name" and "Value" whereas we want "name" and "value" (lowercase).

The error message you are getting `Cannot complete value of unexpected type "EnvVariable."` doesn't seem to indicate that's the problem but worth a shot.

Also since I started writing this comment Alex commented, +1 to his suggestion too.

Reply:
good point. I also think it's unrelated but you're right about the naming.